frontend/src/modules/search/components/SearchBox.js

```diff
@@ -143,6 +144,22 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
         }
     }
 
+    static getDerivedStateFromProps(prevProps: InternalProps, nextProps: Object) {
```

Comment:
This could do with some comments to explain what it's doing and why.

Reply:
After reading the documentation for `getDerivedStateFromProps`, I'm getting wary about using it. Can you please explain precisely what you're achieving here, and maybe we can find a different, simpler way to do it?

Reply:
I just added comments but I have not pushed the code yet.
The local state in that component is delayed to get updated, so after clicking on the status numbers, the status value on the search box placeholder does not get updated  not until one refreshes the page. 
I used the `getDerivedStateFromPops` to ensure the state of the statuses is updated upon changes in the DOM.

Reply:
I couldn't use `componentDidUpdate` to set the new state of statuses because it caused an infinite loop. 
In this function I also need to change the parameter `nextProps` to `state` to be more readable.
The setState called in `componentDidUpdate` does not do anything since `search` key from props is always null. I can use the lifecycle method to update state of the statuses. 
I understand the caveats of using `getDerivedStateFromProps`. Let me see what I can do about it. It should be the last option.

Reply:
@adngdb what's your take on this? 
I have not found another way yet of updating the state of statuses immediately.

Reply:
I thought about it more, and I now think this is actually the correct solution. Congrats on finding it! :-)